### ISSUE TITLE
test hangs: "Output buffer size should be larger than 16 bytes"

### ISSUE THREAD

**Issue by danstender** [Blosc/c-blosc#246]
In Debian we've got test hangs with Blosc 1.14.4 [1] in the depend packages. The autotesting of python-blosc [2] hangs with:

```
decompress (blosc.toplevel) Doctest: blosc.toplevel.decompress ... Output buffer size should be larger than 16 bytes
```
And I have a test hang from working on a new Debian package of bloscpack which might be related to this:

```
test_numpy_io.test_numpy_dtypes_shapes_order(array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]), array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])) ... Output buffer size should be larger than 16 bytes
```

The related Debian bug is: https://bugs.debian.org/913667

[1] build log (i386): https://buildd.debian.org/status/fetch.php?pkg=c-blosc&arch=i386&ver=1.14.4
[2] https://ci.debian.net/packages/p/python-blosc/unstable/amd64/

**Comment by danstender** [comment]
Valentin Haenel pointed this one to be #242.

**Comment by esc** [comment]
The following commit might be the culprit:

https://github.com/Blosc/c-blosc/commit/2f7c6238fada1a6e0f2f9f69717e75b95559e9b6

**Comment by esc** [comment]
I investigated further and used the following pull-request:

https://github.com/Blosc/python-blosc/pull/157

To make the following fix:

https://github.com/Blosc/python-blosc/pull/157/commits/a9ca50f8a03ec5f20e7f8f72417183f5e1a4ee1b

Which resulted in the following green-build:

https://travis-ci.org/esc/python-blosc/builds/457398820

This might just about fix it.

**Comment by esc** [comment]
I staged a pull-request to fix this here:

https://github.com/Blosc/c-blosc/pull/247

**Comment by esc** [comment]
@kalvdans not sure exactly why. Incidentally, I can confirm, that when trying to use pure c-blosc (1.14.4) to compress an empty buffer the test suite hangs too. So here are my current potential culprits:

* Something wrong with the error handling in c-blosc itself.
* Something wriong with the error handing in python-blosc, errors may not propagate correctly
* It might be hanging in python-blosc because it is a doctest
* But bloscpack tests hang too, this isn't a doctest, but an empty Numpy array, so it is most likely an issue in c-blosc somewhere.

**Comment by esc** [comment]
Fixed via #247
